franco/Project/org.fundacionjala.automationproject/src/main/java/org/fundacionjala/automationproject/ui/MainPage.java

```diff
@@ -0,0 +1,100 @@
+package org.fundacionjala.automationproject.ui;
+
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+/**
+ * Created by Franco Aldunate on 11/14/2017.
+ */
+public class MainPage extends BasePage {
+  @FindBy(name = "userName")
+  private WebElement userName;
+
+  @FindBy(name = "password")
+  private WebElement password;
+
+  @FindBy(name = "login")
+  private WebElement logInButton;
+
+  @FindBy(linkText = "REGISTER")
+  private WebElement registerUserLink;
+
+  private String usernameInput;
+  private String passwordInput;
+
+  /**
+   * <p>This constructor receives login field values.</p>
+   */
+  public MainPage() {
+    super();
+  }
+
+  /**
+   * <p>This method sets user name value.</p>
+   *
+   * @param userNameInput is the value of the user name.
+   */
+  public void setUserName(String userNameInput) {
+    this.usernameInput = userNameInput;
+  }
+
+  /**
+   * <p>This method sets password value.</p>
+   *
+   * @param passwordInput is the value of the user password.
+   */
+  public void setPassword(String passwordInput) {
+    this.passwordInput = passwordInput;
+  }
+
+  /**
+   * <p>This method fills user name field.</p>
+   */
+  public void fillUserName() {
+    userName.sendKeys(usernameInput);
+  }
+
+  /**
+   * <p>This method fills user name field.</p>
+   */
+  public void fillPassword() {
+    password.sendKeys(passwordInput);
+  }
+
+  /**
+   * <p>This method clears form fields.</p>
+   */
+  private void clearFields() {
```

Comment:
I think that should user .clear() in the setter of the webElements

Reply:
I used clear() in this method to give the setters just the responsibility to set values, not to do two things.